### PR TITLE
Revert rotate default to before #1022

### LINKED ISSUE #1022: ag.rotate should rotate around selection center of mass/geometry
### Expected behaviour

If I do `ag.rotate(R)` I would expect that I can rotate the protein I selected around it's center of mass/geometry.  I would propose we add a kwarg `origin` for this. This point determines the origin or the rotation and we default to the center of geometry (or lab for backwards compatibility)

``` python
def rotate(self, R, origin='cog'):
    if origin=='cog':
        point = self.center_of_geometry()
    elif origin=='com':
        point = self.center_of_mass()
    elif origin=='lab':
        point = np.zeros(3)

    self.translate(-point)
    # do rotation
    self.translate(point)
```
### Actual behaviour

The center of the rotation is the center of the coordinate system. This means instead of just an internal rotation I can rotate protein in my lab system to another place.
### Currently version of MDAnalysis:

(run `python -c "import MDAnalysis as mda; print(mda.__version__)"`)
0.16.0dev

### PR DESCRIPTION
**Pull request by kain88-de**
Fixes #1022

Changes made in this Pull Request:
 - `ag.rotate` rotates around 0, 0, 0 again.

This version now is good. It keeps old code working and it allows me to quickly check the center of rotation and change it.

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: testsuite/MDAnalysisTests/datafiles.py [hunk 1]
```diff
@@ -362,5 +364,8 @@
 MMTF = resource_filename(__name__, 'data/173D.mmtf')
 MMTF_gz = resource_filename(__name__, 'data/5KIH.mmtf.gz')
 
+ALIGN_BOUND = resource_filename(__name__, 'data/analysis/align_bound.pdb.gz')
+ALIGN_UNBOUND = resource_filename(__name__, 'data/analysis/align_unbound.pdb.gz')
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/datafiles.py]: Are there any other two protein datafiles we already have and could use for this test? I didn't find any skipping over the list.

File: package/MDAnalysis/core/groups.py [hunk 2]
```diff
@@ -672,34 +672,29 @@ def translate(self, t):
...
-        order to perform a rotation around, say, the origin, use
-        ``point=[0, 0, 0]``.
-
-        :math:`\mathsf{R}` is a 3x3 orthogonal matrix that transforms a vector
-        :math:`\mathbf{x} \rightarrow \mathbf{x}'`:
-
-        .. math::
-
-            \mathbf{x}' = \mathsf{R}\mathbf{x}
+        By default rotates around center of origin ``point=(0, 0, 0)``.
```
Comment (richardjgowers) [inline package/MDAnalysis/core/groups.py]: Would it be worth adding a comment how to rotate around the COM of the AG as a hint?  Ie how to get the behaviour that you wanted to implement as default

Comment (kain88-de) [inline package/MDAnalysis/core/groups.py]: I added in the notes now how once can chose a different rotation center. For my original case it is already good that the paramter list gives me the option and tells me the default is 0, 0, 0

**Comment by orbeckst** [comment]
Can I merge?

**Comment by kain88-de** [comment]
Sure.

**Pull request merged**